### PR TITLE
Add a scripted test for caliban-codegen-sbt

### PR DESCRIPTION
**Pull request by xirc**
Add a scripted test that verifies `caliban-codegen-sbt` generated code compiles or not.

As you may know, a scripted test can test a `sbt-plugin`.
See https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html for details.

This PR includes
- Add a scripted test that verifies `caliban-codegen-sbt` generated code compiles or not
- Add help for scripted tests to sbt console
- Enable scripted tests in CI pipeline

Unfortunately, I'm not a GraphQL expert, so I'm not sure that added tests cover all (or enough) GraphQL syntax.

The GraphQL schema written in `codegen-sbt/src/sbt-test/codegen/test-compile/project/schema.graphql` is picked up from
https://github.com/ghostdogpr/caliban/blob/master/tools/src/test/scala/caliban/tools/ClientWriterSpec.scala .

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
       - checkout
       - restore_cache:
           key: sbtcache
-      - run: sbt ++2.13.3! core/test http4s/compile akkaHttp/compile finch/compile play/compile examples/compile catsInterop/compile monixInterop/compile tapirInterop/test clientJVM/test federation/test
+      - run: sbt ++2.13.3! core/test http4s/compile akkaHttp/compile finch/compile play/compile examples/compile catsInterop/compile monixInterop/compile tapirInterop/test clientJVM/test federation/test codegenSbt/scripted
```
Comment (ghostdogpr) [inline .circleci/config.yml:38]: I think the sbt plugin only compiles with 2.12

**Review by ghostdogpr — COMMENTED** [review]
Wow, this is awesome, I had no idea we could do that! 👏

File: .circleci/config.yml (hunk 1 above)
Comment (xirc) [reply .circleci/config.yml]: Thanks!
I've changed it. I hope the CI will be success.

**Review by ghostdogpr — APPROVED** [review]
It's really nice. Now if we find a schema that causes compile errors we can add it as a regression test. Thanks a lot.

**Comment by ghostdogpr** [comment]
@xirc I noticed something weird. Since this PR it seems that a snapshot of `caliban-codegen-sbt` is published at every build, including on branches (normally snapshots are published on master only). For example in https://app.circleci.com/pipelines/github/ghostdogpr/caliban/1757/workflows/58347ec5-d4cf-4c7c-82e9-1ee660921b37/jobs/7172:
```
[info] Main Scala API documentation successful.
[info] :: delivering :: com.github.ghostdogpr#caliban-codegen-sbt;0.9.3+23-48b8e3c8-SNAPSHOT :: 0.9.3+23-48b8e3c8-SNAPSHOT :: integration :: Mon Nov 23 00:52:59 UTC 2020
[info] 	delivering ivy file to /home/circleci/project/codegen-sbt/target/scala-2.12/sbt-1.0/ivy-0.9.3+23-48b8e3c8-SNAPSHOT.xml
[info] 	published caliban-codegen-sbt to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-codegen-sbt/scala_2.12/sbt_1.0/0.9.3+23-48b8e3c8-SNAPSHOT/poms/caliban-codegen-sbt.pom
[info] 	published caliban-codegen-sbt to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-codegen-sbt/scala_2.12/sbt_1.0/0.9.3+23-48b8e3c8-SNAPSHOT/jars/caliban-codegen-sbt.jar
[info] 	published caliban-codegen-sbt to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-codegen-sbt/scala_2.12/sbt_1.0/0.9.3+23-48b8e3c8-SNAPSHOT/srcs/caliban-codegen-sbt-sources.jar
[info] 	published caliban-codegen-sbt to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-codegen-sbt/scala_2.12/sbt_1.0/0.9.3+23-48b8e3c8-SNAPSHOT/docs/caliban-codegen-sbt-javadoc.jar
[info] 	published ivy to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-codegen-sbt/scala_2.12/sbt_1.0/0.9.3+23-48b8e3c8-SNAPSHOT/ivys/ivy.xml
[info] Main Scala API documentation successful.
[info] :: delivering :: com.github.ghostdogpr#caliban-tools_2.12;0.9.3+23-48b8e3c8-SNAPSHOT :: 0.9.3+23-48b8e3c8-SNAPSHOT :: integration :: Mon Nov 23 00:53:01 UTC 2020
[info] 	delivering ivy file to /home/circleci/project/tools/target/scala-2.12/ivy-0.9.3+23-48b8e3c8-SNAPSHOT.xml
[info] 	published caliban-tools_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-tools_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/poms/caliban-tools_2.12.pom
[info] 	published caliban-tools_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-tools_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/jars/caliban-tools_2.12.jar
[info] 	published caliban-tools_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-tools_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/srcs/caliban-tools_2.12-sources.jar
[info] 	published caliban-tools_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-tools_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/docs/caliban-tools_2.12-javadoc.jar
[info] 	published ivy to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-tools_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/ivys/ivy.xml
[info] Main Scala API documentation successful.
[info] :: delivering :: com.github.ghostdogpr#caliban-client_2.12;0.9.3+23-48b8e3c8-SNAPSHOT :: 0.9.3+23-48b8e3c8-SNAPSHOT :: integration :: Mon Nov 23 00:53:01 UTC 2020
[info] 	delivering ivy file to /home/circleci/project/client/.jvm/target/scala-2.12/ivy-0.9.3+23-48b8e3c8-SNAPSHOT.xml
[info] 	published caliban-client_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-client_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/poms/caliban-client_2.12.pom
[info] 	published caliban-client_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-client_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/jars/caliban-client_2.12.jar
[info] 	published caliban-client_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-client_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/srcs/caliban-client_2.12-sources.jar
[info] 	published caliban-client_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-client_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/docs/caliban-client_2.12-javadoc.jar
[info] 	published ivy to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban-client_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/ivys/ivy.xml
model contains 250 documentable templates
[info] Main Scala API documentation successful.
[info] :: delivering :: com.github.ghostdogpr#caliban_2.12;0.9.3+23-48b8e3c8-SNAPSHOT :: 0.9.3+23-48b8e3c8-SNAPSHOT :: integration :: Mon Nov 23 00:53:12 UTC 2020
[info] 	delivering ivy file to /home/circleci/project/core/target/scala-2.12/ivy-0.9.3+23-48b8e3c8-SNAPSHOT.xml
[info] 	published caliban_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/poms/caliban_2.12.pom
[info] 	published caliban_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/jars/caliban_2.12.jar
[info] 	published caliban_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/srcs/caliban_2.12-sources.jar
[info] 	published caliban_2.12 to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/docs/caliban_2.12-javadoc.jar
[info] 	published ivy to /home/circleci/.ivy2/local/com.github.ghostdogpr/caliban_2.12/0.9.3+23-48b8e3c8-SNAPSHOT/ivys/ivy.xml
[info] Running 1 / 1 (100.00%) scripted tests with RunFromSourceMain
[info] Running codegen/test-compile
```
Any idea why?

**Comment by ghostdogpr** [comment]
Ah it's only locally, it should be okay then.

**Comment by xirc** [comment]
@ghostdogpr 
Sorry for the late response. The scripted test that I added, invokes `publishLocal`.
Scripted tests need to resolve `caliban-codegen-sbt`, etc... from **local** artifacts repository, and run their tests.
This is why `publishLocal` is invoked.

I think `publishLocal` doesn't publish SNAPSHOT artifacts into your **public** artifacts repository.
But, if you noticed something else went wrong, please let me know.
And, of course, any time, we can disable the scripted test in the CI pipeline by changing https://github.com/ghostdogpr/caliban/blob/master/.circleci/config.yml#L24.

**Comment by ghostdogpr** [comment]
@xirc yep, this is fine. Thank you!

**Pull request merged**